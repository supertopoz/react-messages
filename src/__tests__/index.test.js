// @flow

import React from 'react'
import ReactMessages from '../index'
import { shallow } from 'enzyme'

describe('<ReactMessages />', () => {
  it('should render itself', () => {
    const message = 'Just a fake message'
    const next = true
    const component = shallow(<ReactMessages message={message} next={next} />)
    expect(component).toMatchSnapshot()
  })

  it('should render the right message', () => {
    const message = 'Just a fake message'
    const next = true
    const component = shallow(<ReactMessages message={message} next={next} />)
    expect(component.text()).toEqual(message)
  })

  it('should render the right icon', () => {
    const message = 'Just a fake message'
    const next = true
    const icons = ['heart', 'warning', 'info', 'thumbs-up', 'thumbs-down']
    const random = Math.floor(Math.random() * icons.length)
    const component = shallow(<ReactMessages message={message} next={next} icon={icons[random]} />)
    expect(component.find(`.icon-rm-${icons[random]}`).length).toEqual(1)
  })

  it('should return null when the prop next is false', () => {
    const message = 'Just a fake message'
    const next = false
    const component = shallow(<ReactMessages message={message} next={next} />)
    expect(component.find('.msg-wrapper').length).toEqual(0)
  })
})