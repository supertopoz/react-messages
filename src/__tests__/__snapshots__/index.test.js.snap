// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReactMessages /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReactMessages
    duration={2000}
    error={false}
    icon="heart"
    message="Just a fake message"
    next={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="rm-item "
      >
        <span
          className="rm-txt icon-rm-heart"
        />
        <span
          className="rm-txt"
        >
          Just a fake message
        </span>
      </p>,
      "className": "rm-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="rm-txt icon-rm-heart"
          />,
          <span
            className="rm-txt"
          >
            Just a fake message
          </span>,
        ],
        "className": "rm-item ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rm-txt icon-rm-heart",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Just a fake message",
            "className": "rm-txt",
          },
          "ref": null,
          "rendered": "Just a fake message",
          "type": "span",
        },
      ],
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="rm-item "
        >
          <span
            className="rm-txt icon-rm-heart"
          />
          <span
            className="rm-txt"
          >
            Just a fake message
          </span>
        </p>,
        "className": "rm-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="rm-txt icon-rm-heart"
            />,
            <span
              className="rm-txt"
            >
              Just a fake message
            </span>,
          ],
          "className": "rm-item ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "rm-txt icon-rm-heart",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Just a fake message",
              "className": "rm-txt",
            },
            "ref": null,
            "rendered": "Just a fake message",
            "type": "span",
          },
        ],
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
